src/main/java/uk/ac/ebi/embl/api/validation/dao/EraproDAOUtilsImpl.java

```diff
@@ -392,10 +427,39 @@ public boolean isProjectValid(String project) throws SQLException
 		}
 	}
 
+	/**
+	 * Given SRA sample ID is used to look up corresponding Biosample accession which then is used to retreive sample
+	 * information from Biosamples first. If the sample could not be retrieved from Biosamples then it is loaded
+	 * from ERAPRO.
+	 *
+	 * @param sampleId SRA sample accession.
+	 * @return
+	 * @throws Exception
+	 */
 	@Override
 	public SourceFeature getSourceFeature(String sampleId) throws Exception {
```

Comment:
There are two methods in EraproDAOUtilsImpl that are used to create source feature using sample information:
- getSourceFeature
- createMasterEntry

What is common to both is that they call 
```
private SampleEntity getSampleAttributes( String sampleId)
```

to get extract sample attributes from the sample. 

- I would expect ```getSampleAttributes``` to be changed so that sample information is not only retrieved from ERA oracle.
- I would also expect getSourceFeature and createMasterEntry only changed minimally or not al all when they call ```getSampleAttributes```.